File: appli-android/app/src/main/java/com/example/sportapp/designsystem/common_components/CustomSpacer.kt
package com.example.sportapp.designsystem.common_components

import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxHeight
import androidx.compose.foundation.layout.width
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp

/**
 * Spacer vertical de largeur paramétrable (default 6dp), qui remplit la hauteur du parent.
 * Créé pour ajouter un gap horizontal entre cellules dans une Row.
 *
 * Renommé depuis l'ancien `CustomVerticalDivider` (qui était mal nommé : ce n'est pas un
 * divider mais un spacer transparent — le param `color` était d'ailleurs ignoré).
 */
@Composable
fun CustomSpacer(width: Dp = 6.dp) {
    Spacer(modifier = Modifier.width(width).fillMaxHeight())
}

File: appli-web/src/app/designsystem/common_components/custom-spacer.ts
import { ChangeDetectionStrategy, Component, input } from '@angular/core';

/**
 * Spacer horizontal de largeur paramétrable (default 6px), remplit la hauteur du parent.
 * Miroir de CustomSpacer.kt (ex-CustomVerticalDivider, renommé côté Android car c'est un spacer
 * transparent, pas un divider).
 */
@Component({
  selector: 'app-custom-spacer',
  changeDetection: ChangeDetectionStrategy.OnPush,
  template: '',
  host: {
    '[style.display]': '"inline-block"',
    '[style.alignSelf]': '"stretch"',
    '[style.width.px]': 'width()',
  },
})
export class CustomSpacer {
  readonly width = input(6);
}
